contracts/tokenlock/TokenLock.sol

```diff
@@ -190,17 +191,27 @@ abstract contract TokenLock is Ownable, ITokenLock {
         if (!revocable)
             revert LockIsNonRevocable();
 
-        uint256 unvestedAmount = managedAmount - vestedAmount();
+        if (isRevoked)
+            revert LockIsAlreadyRevoked();
+
+        uint256 vestedAmount = vestedAmount();
+
+        uint256 unvestedAmount = managedAmount - vestedAmount;
         if (unvestedAmount == 0)
             revert NoAvailableUnvestedAmount();
 
         isRevoked = true;
 
+        managedAmount = vestedAmount;
```

Comment:
resetting managedAmount will also lower amountPerPeriod(), and with that it will also make availableAmount() and vestedAmount() smaller.

So that means that this happens (suppose user has not withdrawn any funds yet, managedAmount is 100, there are 10 periods, of which 5 have passed)
1. owner calls revoke(), withdraws 50 tokens, and sets managedAmount to 50
2. beneficiary calls release() but will get only 25 tokens

at least I think this is the case, isnt it?

Reply:
I guess @bahurum 's suggestion (update relasedAmount, not manageedAmount) may avoid this problem

Reply:
We set endTime = block.timestamp; so that all periods are passed. So the full amount is ready to be released.

Reply:
In my comment above, I did not notice this line, sorry. With this line the accounting is ok.

Reply:
Oh @ben-kaufman sorry about that you are right of course